sympy/stats/rv_interface.py

```diff
@@ -422,6 +422,9 @@ def median(X, evaluate=True, **kwargs):
     .. [1] https://en.wikipedia.org/wiki/Median#Probability_distributions
 
     """
+    if not is_random(X):
+        return X
```

Comment:
Is it necessary to special case this? What fails otherwise (and can the other thing that fails be fixed instead)?

Reply:
The problem is that `pspace` cannot be called on an expression with no random variables. It would raise `ValueError: Expression containing Random Variable expected`. One alternative to handling it as a special case would be to let `pspace` return something useful for constant expressions. Perhaps something like a "Dirac distribution pspace", "constant distribution pspace" or "degenerate distribution pspace". Perhaps this would remove the need for special cases also in other situations.

Reply:
Okay, so that goes back to what you originally suggested about having a Dirac distribution. Right now a constant expression needs special case handling to work in some situations. However making it work in general requires implementing a pspace. Maybe we should avoid adding special case code if there is a more general way to fix it.

Reply:
I'd assume adjusting `pspace` like that would simply move the `if` statement around and wouldn't actually remove it. Removing similar statements from higher-level functions might create bugs with unconsidered edge-cases.

Reply:
I think adding a new pspace is the right thing to do. That being said I don't have the full overview of the stats module so I cannot tell if it would directly affect more code than this. One way of finding out is to try it and ask the unit tests.

Reply:
> I'd assume adjusting `pspace` like that would simply move the `if` statement around and wouldn't actually remove it.

The question is whether we add a special case to every function that does anything with random variables or whether we implement the trivial pspace so all of those functions can work without needing to care that the expression is constant or not.

I don't think that this needs to hold up the PR but adding more checks like this to every function that doesn't work properly with constant RVs would not be the right approach.